distributed/tests/test_failed_workers.py

```diff
@@ -465,20 +465,23 @@ async def test_restart_timeout_on_long_running_task(c, s, a):
 async def test_worker_time_to_live(c, s, a, b):
     from distributed.scheduler import heartbeat_interval
 
-    # worker removal is also controlled by 10 * heartbeat
     assert set(s.workers) == {a.address, b.address}
-    interval = 10 * heartbeat_interval(len(s.workers)) + 0.5
 
     a.periodic_callbacks["heartbeat"].stop()
-    await asyncio.sleep(0.010)
-    assert set(s.workers) == {a.address, b.address}
+    while a.heartbeat_active:
+        await asyncio.sleep(0.01)
 
     start = time()
     while set(s.workers) == {a.address, b.address}:
-        await asyncio.sleep(interval)
-        assert time() < start + interval + 0.1
-
-    set(s.workers) == {b.address}
+        await asyncio.sleep(0.01)
+    assert set(s.workers) == {b.address}
```

Comment:
This seems to improve upon the previous version by:

1. Waiting for worker `a` heartbeat to stop
2. Waiting for the scheduler to drop worker `a`.
3. Upping the tolerance level for the timeout.